packages/server/src/index.ts

```diff
@@ -402,58 +466,84 @@ export class App {
         // ----------------------------------------
 
         // Get all chatmessages from chatflowid
-        this.app.get('/api/v1/chatmessage/:id', async (req: Request, res: Response) => {
-            const sortOrder = req.query?.order as string | undefined
-            const chatId = req.query?.chatId as string | undefined
-            const memoryType = req.query?.memoryType as string | undefined
-            const sessionId = req.query?.sessionId as string | undefined
-            const startDate = req.query?.startDate as string | undefined
-            const endDate = req.query?.endDate as string | undefined
-            let chatTypeFilter = req.query?.chatType as chatType | undefined
-
-            if (chatTypeFilter) {
-                try {
-                    const chatTypeFilterArray = JSON.parse(chatTypeFilter)
-                    if (chatTypeFilterArray.includes(chatType.EXTERNAL) && chatTypeFilterArray.includes(chatType.INTERNAL)) {
-                        chatTypeFilter = undefined
-                    } else if (chatTypeFilterArray.includes(chatType.EXTERNAL)) {
-                        chatTypeFilter = chatType.EXTERNAL
-                    } else if (chatTypeFilterArray.includes(chatType.INTERNAL)) {
-                        chatTypeFilter = chatType.INTERNAL
+        this.app.get(
+            '/api/v1/chatmessage/:id',
+            query('chatId').notEmpty().escape(),
```

Comment:
all these query are optional. by declaring this as `notEmpty()`, a basic API call to get chat messages without these query will fail

Reply:
I have changed the entire implementation and removed validation for each param